src/cluster_legacy.c

```diff
@@ -1252,6 +1252,57 @@ void clusterInitLast(void) {
 
 /* Called when a cluster node receives SHUTDOWN. */
 void clusterHandleServerShutdown(void) {
+    if (nodeIsPrimary(myself) && server.auto_failover_on_shutdown) {
+        /* Find the first best replica, that is, the replica with the largest offset. */
+        int old_replica = 0;
+        client *best_replica = NULL;
+        listIter replicas_iter;
+        listNode *replicas_list_node;
+        listRewind(server.replicas, &replicas_iter);
+        while ((replicas_list_node = listNext(&replicas_iter)) != NULL) {
+            client *replica = listNodeValue(replicas_list_node);
+            /* This is done only when the replica offset is caught up, to avoid data loss.
+             * And 0x80100 is 8.1.0, we only support this feature in this version. */
+            if (replica->repl_data->replica_version < 0x80100) {
+                old_replica = 1;
+                best_replica = NULL;
+                break;
+            }
+            if (replica->repl_data->repl_state == REPLICA_STATE_ONLINE &&
+                replica->name && sdslen(replica->name->ptr) == CLUSTER_NAMELEN &&
```

Comment:
not sure I understand the need for this name check? also isn't `client setname` a best effort operation?

Reply:
we will need the name to send the failover command, it rely a replicaid arg.